### DISCUSSION TITLE
renovatebot merging unrelated pull requests

### DISCUSSION THREAD

**Discussion by nijel** [renovatebot/renovate#29422] [Request Help] [answered]
### What would you like help with?

I think I found a bug

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

Today, rennovatebot merged a pull request submitted by someone else:

![image](https://github.com/renovatebot/renovate/assets/212189/68fda42e-9fda-4ff6-a233-34bded28cd93)

You can find it at https://github.com/WeblateOrg/weblate/pull/11510#event-13028605585. It was updated to be blank just before that, so no code was changed, but still this worries me.

### Logs (if relevant)

I can't find anything related in the log of run at the time pull request has been merged: https://developer.mend.io/github/WeblateOrg/weblate/-/job/018fe08b-ea4c-79fb-8000-e7b7cc7bb29f

**Comment by rarkins** [comment, marked as answer]
I think this is a quirk/bug from GitHub and not something Renovate did specifically. It's likely that the emptyness of the PR is the _reason_ GitHub reported it this way.

At this point the PR became empty/invalid/pointless: 
<img width="681" alt="image" src="https://github.com/renovatebot/renovate/assets/6311784/7e4ac67e-579c-4d96-b603-02287009b62a">

It was matching this main branch commit:
<img width="841" alt="image" src="https://github.com/renovatebot/renovate/assets/6311784/8e56db9d-f725-4a92-97b3-5c4b73a9835b">

GitHub generally closes PRs once they have no change, but perhaps the fact this was force pushed or was on a fork meant it didn't happen then.

Afterwards Renovate's own PR merged:
<img width="543" alt="image" src="https://github.com/renovatebot/renovate/assets/6311784/93bb8c28-567d-4ee8-b23a-523e680bdc5c">

This likely triggered a "cleanup" by GitHub to close the empty PR and it was (incorrectly) marked by GitHub as _merged_ by Renovate because Renovate's PR included the commit SHA. Compare the times:

<img width="511" alt="image" src="https://github.com/renovatebot/renovate/assets/6311784/6abc1daf-1a9c-4f10-bb9a-e46b5645b212">

<img width="691" alt="image" src="https://github.com/renovatebot/renovate/assets/6311784/3acf7a58-725d-4144-9f48-6cb61ceb9e8d">

**Reply by nijel** [reply]
Thanks for investigating this, that kind of makes sense, but I've posted it at https://github.com/orgs/community/discussions/127200 to get feedback from GitHub as well.
